go/pkg/prometheus/metrics/keys.go

```diff
@@ -0,0 +1,42 @@
+/*
+Package metrics provides Prometheus metric collectors for monitoring application performance.
+
+This file contains Key-Verification-related metrics for tracking what keys do.
+*/
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	// KeyVerificationsTotal tracks the number of key verifications handled, labeled by some data.
+	// Use this counter to monitor API traffic patterns and error rates.
+	//
+	// Example usage:
+	//   metrics.VerificationsTotal.WithLabelValues("GET", "/users", "200").Inc()
+	KeyVerificationsTotal = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Subsystem:   "key",
+			Name:        "verifications_total",
+			Help:        "Total number of Key verifications processed.",
+			ConstLabels: constLabels,
+		},
+		[]string{"workspaceId", "apiId", "keyId", "valid", "code"},
+	)
+
+	// KeyCreditsSpentTotal tracks the total credits spent by keys, labeled by workspace ID, key ID, and identity ID.
+	// Use this counter to monitor credit usage patterns and error rates.
+	//
+	// Example usage:
+	//   metrics.CreditsSpentTotal.WithLabelValues("workspace1", "key1", "identity1").Inc()
```

Comment:
this example seems off, I assume 'deducted' is supposed to be a boolean? but the example omits it and then uses Inc() whereas we probaably use this with a custom cost